### PR TITLE
added drone

### PR DESCRIPTION
**Pull request by frosthamster**

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+
+WORKDIR /opt/drone
+
+RUN apk add --update \
+    bash \
+    curl \
+    && rm -rf /var/cache/apk/*
+
+COPY go.mod go.sum ./
+RUN go mod download && go mod verify
```
Comment (cauchy2384) [inline Dockerfile:11]: Wdyt about using multi-stage build?

File: src/drone/main.go [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+func main() {
+	ctx, stop := signal.NotifyContext(context.Background(), os.Interrupt)
+	defer stop()
+
+	cfg := Config{
+		TgKey:       os.Getenv("DRONE_TG_BOT_API_KEY"),
+		LCDailyCron: os.Getenv("DRONE_LC_DAILY_CRON"),
+	}
+	if err := StartDrone(ctx, cfg); err != nil {
+		panic(err)
```
Comment (cauchy2384) [inline src/drone/main.go:20]: Have you considered using print to the stderr & setting os.code instead?

File: src/drone/bot.go [hunk 3]
```diff
@@ -0,0 +1,81 @@
...
+		return err
+	}
+	dr := drone{bot: bot}
+
+	scheduler, err := gocron.NewScheduler(gocron.WithLocation(time.UTC))
+	if err != nil {
+		return err
+	}
+
+	if len(cfg.LCDailyCron) == 0 {
```
Comment (cauchy2384) [inline src/drone/bot.go:33]: Wdyt about moving default values setting closer to the config parsing?

File: src/tg/boar_d_white.go [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+import (
+	"fmt"
+
+	tele "gopkg.in/telebot.v3"
+)
+
+const (
+	DefaultDailyHeader = "LeetCode Daily Question"
+
+	boarDWhiteChatID           = tele.ChatID(-1001640461540)
```
Comment (cauchy2384) [inline src/tg/boar_d_white.go:12]: You definitely want to move this to config as we code a mad owner who might be closing chats and topics every now and then :D

File: src/tg/boar_d_white.go [hunk 5]
```diff
@@ -0,0 +1,34 @@
...
+	boarDWhiteLeetCodeThreadID = 10095
+)
+
+func SendLCDailyToBoarDWhite(bot *tele.Bot, header, dailyLink string) error {
+	payload := fmt.Sprintf("%v\n%v", header, dailyLink)
```
Comment (cauchy2384) [inline src/tg/boar_d_white.go:17]: Is there a specific reason to use `%v` instead of `%s`?

Comment (frosthamster) [reply src/tg/boar_d_white.go:17]: it's 0iq move to not remember all combinations of cryptic %d %s %...
and use %v %+v everywhere instead

File: src/leetcode/daily.go [hunk 6]
```diff
@@ -0,0 +1,60 @@
...
+		}
+
+		resp := dailyQuestionResp{}
+		if err := json.Unmarshal(out, &resp); err != nil {
+			return "", err
+		}
+
+		link := resp.Data.ActiveDailyCodingChallengeQuestion.Link
+		if len(link) == 0 {
+			return "", errors.New("got empty link")
```
Comment (cauchy2384) [inline src/leetcode/daily.go:51]: It is part of the package API therefore must be a global variable.

File: src/retry/retry.go [hunk 7]
```diff
@@ -0,0 +1,59 @@
...
+type break_ struct {
+	error
+}
+
+func (break_) Is(err error) bool {
+	_, ok := err.(break_)
+	return ok
+}
+
+func Break(err error) error {
```
Comment (cauchy2384) [inline src/retry/retry.go:18]: Am I missing something or is it unused?

File: src/retry/retry.go [hunk 8]
```diff
@@ -0,0 +1,59 @@
+package retry
+
+import (
+	"context"
+	"errors"
+	"time"
+)
+
+type break_ struct {
```
Comment (cauchy2384) [inline src/retry/retry.go:9]: You could make it simpler like
```
var ErrBreak = errors.New("break")

func Break(err error) error {
    return fmt.Errorf("%w: %w", ErrBreak, err)
}

```

**Review by cauchy2384 — APPROVED** [review]
Great job, thank you for taking the initiative!

File: src/retry/retry.go (hunk 8 above)
Comment (frosthamster) [reply src/retry/retry.go:9]: oh, good point, i forgot to unwrap the error, i don't want to leak any retry logic to code that uses it
it's intended to be uses just as marker for a particular error

**Pull request merged**